util/dockerfiles/devcontainer/Dockerfile

```diff
@@ -41,13 +42,33 @@
 # if the container is used for education or demonstration purposes.
 FROM --platform=${BUILDPLATFORM} ghcr.io/gem5/ubuntu-24.04_all-dependencies:latest as builder
 RUN git clone --branch v24.0 https://github.com/gem5/gem5
+
 WORKDIR /gem5
 RUN scons build/ALL/gem5.opt -j`nproc`
+RUN scons defconfig build/ALL_CHI build_opts/ALL && \
+    scons setconfig build/ALL_CHI RUBY_PROTOCOL_CHI=y && \
+    scons build/ALL_CHI/gem5.opt -j`nproc`
+RUN scons build/VEGA_X86/gem5.opt -j`nproc`
 
-# Stage 2: The final stage where we create the devcontainer image.
+# Stage 2: Create the devcontainer image.
 # This includes all dependencies for building and running gem5, the
 # dependencies for developing gem5, and a pre-built gem5 binary.
 FROM --platform=${BUILDPLATFORM} ghcr.io/gem5/ubuntu-24.04_all-dependencies:latest
+COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5-mesi
+COPY --from=builder /gem5/build/ALL_CHI/gem5.opt /usr/local/bin/gem5
+COPY --from=builder /gem5/build/VEGA_X86/gem5.opt /usr/local/bin/gem5-vega
+
+# Stage 3: Install additional packages.
+# While the base image includes all dependencies for building and running gem5,
+# we install packages which are useful when running in a dev container and/or
+# developing gem5. (e.g. vim to edit files, qemu for building disk images,
+# etc.).
 RUN apt -y update  && \
-    apt -y install vim
-COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5
+    apt -y install \
+    vim \
+    qemu-system \
+    qemu-kvm \
+    g++-aarch64-linux-gnu \
+    g++-riscv64-linux-gnu \
+    gcc-aarch64-linux-gnu \
+    gcc-riscv64-linux-gnu
```

Comment:
Can you add the following code? I'm not 100% sure what should go here vs in our startup script. This is necessary to get `menuconfig` to work.

```
apt-get update && apt-get install locales
locale-gen en_US.UTF-8
export LANG="en_US.UTF-8"
```